packages/frontend/src/pages/Dashboard.tsx

```diff
@@ -41,9 +44,26 @@ const GridTile: FC<
     >
 > = memo((props) => {
     const { tile } = props;
+
+    const savedChartUuid: string | undefined =
+        tile.type === DashboardTileTypes.SAVED_CHART
+            ? tile.properties?.savedChartUuid || undefined
+            : undefined;
+    const { data: savedQuery, isLoading } = useSavedQuery({
+        id: savedChartUuid,
+    });
     switch (tile.type) {
         case DashboardTileTypes.SAVED_CHART:
-            return <ChartTile {...props} tile={tile} />;
+            if (isLoading) return <></>;
+            return (
+                <UnderlyingDataProvider
+                    filters={savedQuery?.metricQuery.filters}
+                    tableName={savedQuery?.tableName || ''}
+                >
+                    <ChartTile {...props} tile={tile} />
+                    <UnderlyingDataModal />
+                </UnderlyingDataProvider>
+            );
```

Comment:
What you think if we moved this code to `packages/frontend/src/components/DashboardTiles/DashboardChartTile.tsx` where we already load the the saved query and handle the loading state ?

For that we would need to stop using the `useUnderlyingDataContext` in there 🤔 What if we moved the context menu code to a separate component ?

Reply:
Yes, that was my first approach, but that was not working because of the context... I'm not sure which solution is less messy..

Reply:
👍 yup, let's leave it for now. `DashboardChartTile` is becoming massive and needs a refactor soon.